cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/datastore/DatastoreImpl.java

```diff
@@ -656,14 +591,17 @@ private DocumentRevision updateDocumentBody(SQLDatabase db, String docId,
         this.validateDBBody(body);
         CouchUtils.validateRevisionId(prevRevId);
 
+        // TODO quicker way of finding if current than fetching whole revision
         DocumentRevision preRevision = new GetDocumentCallable(docId, prevRevId, this.attachmentsDir, this.attachmentStreamFactory).call(db);
 
         if (!preRevision.isCurrent()) {
             throw new ConflictException("Revision to be updated is not current revision.");
         }
 
         new SetCurrentCallable(preRevision.getSequence(), false).call(db);
-        String newRevisionId = this.insertNewWinnerRevision(db, body, preRevision);
+        InsertRevisionCallable insertRevisionCallable = this.insertNewWinnerRevisionAdaptor(body, preRevision);
+        String newRevisionId = insertRevisionCallable.revId;
+        insertRevisionCallable.call(db);
         return new GetDocumentCallable(docId, newRevisionId, this.attachmentsDir, this.attachmentStreamFactory).call(db);
```

Comment:
Should this be a TODO as well to build the new `DocumentRevision` instead of retrieving the whole document again?